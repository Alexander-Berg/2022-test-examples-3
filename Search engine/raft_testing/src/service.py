import os
import sys
from pysyncobj import replicated_sync

from search.horadric2.proto.structures import unistat_pb2

from search.martylib.raft import BaseRaftConsumer, traced_replicated, exception_handler
from search.martylib.unistat.utils import clean_data, merge_data
from search.martylib.modules import ServiceAndLoopModule
from search.martylib.threading_utils.interval import Interval

from search.raft_testing.services.raft import RaftInterface
from search.raft_testing.proto import structures_pb2


class ReplicatedList(BaseRaftConsumer):

    def __init__(self):
        super().__init__('main-list')
        self.storage = []

    @exception_handler
    @replicated_sync
    @traced_replicated
    def append(self, data):
        self.storage.append(data)

    @exception_handler
    @replicated_sync
    @traced_replicated
    def pop(self):
        self.storage.pop()

    @exception_handler
    @replicated_sync
    @traced_replicated
    def clear(self):
        self.storage.clear()


class RaftTester(RaftInterface):

    storage = ReplicatedList()

    def __init__(self):
        self.loop = Interval(sleep_interval=1, target=lambda: self.add_random_data(structures_pb2.AddRandomDataRequest(), None))

    def add_random_data(self, request, context):
        request.size = request.size or 1024

        self.storage.append(os.urandom(request.size), timeout=30)

    def clear_data(self, request, context):
        self.storage.clear()

    def unistat(self, request=None, context=None):
        data = super().unistat(request, context)
        additional_data = unistat_pb2.UnistatData()

        additional_data.numerical['data-size_ammv'] = sum(map(len, self.storage.storage))

        return merge_data(data, clean_data(additional_data))


class RaftTesterModule(ServiceAndLoopModule):

    @classmethod
    def get_loops(cls):
        yield cls._get_service().loop

    @classmethod
    def get_service(cls):
        return RaftTester()
