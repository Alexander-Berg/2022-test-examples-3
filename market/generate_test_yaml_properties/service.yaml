java_service:
  service_name: framework-test-service
  deploy_type: nanny

trace:
  module: FRAMEWORK_TEST_SERVICE
