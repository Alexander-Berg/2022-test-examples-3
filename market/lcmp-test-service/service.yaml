java_service:
  latest_mj_version: true
  service_name: lcmp-test-service
  root_package: ru.yandex.market.lcmptestservice
  deploy_type: nanny

trace:
  module: MARKET_LCMP_TEST_SERVICE

modules:
  experiments3:
    port: 11923
    consumer: lcmp-test-service
    configSchemas:
      - market/dev-exp/services/lcmp-test-service/exp3_configs/features
      - market/dev-exp/services/lcmp-test-service/exp3_configs/general

#clients:
#  fault_tolerance_presets:
#    circuit_breakers:
#      test_service:
#        failure_rate_threshold: 50
#        permitted_number_of_calls_in_half_open_state: 10
#        sliding_window_size: 100
#        sliding_window_type: COUNT_BASED
#        minimum_number_of_calls: 100
#        writable_stack_trac_enabled: true
#        automatic_transition_from_open_to_half_open_enabled: false
#        wait_interval_function_in_open_state_seconds: 60
#        slow_call_rate_threshold: 100
#        slow_call_duration_threshold_seconds: 60
#        max_wait_duration_in_half_open_state_seconds: 0
#    retries:
#      test_service:
#        max_attempts: 3
#        fail_after_max_attempts: false
#        writable_stack_trace_enabled: true
#        wait_duration_millis: 3000
#
#  list:
#    - openapi_spec_path: market/dev-exp/services/lcmp-test-service/src/main/resources/openapi/api/api.yaml
#      circuit_breaker_id: test_service
#      retry_id: test_service

#tvm:
#  sources:
#    - 2029758
#  destinations:
#    - 2029758
#  env:
#    local:
#      serverTvmDisabled: false
#      clientsTvmDisabled: true
#    production:
#      id: 8888
#      sources:
#        - 32343
#        - 645647
#      destinations:
#        - 423
#        - 224
#    testing:
#      id: 7777
#      sources:
#        - 424
#        - 647
#      destinations:
#        - 42213
#        - 2243
#    prestable:
#      id: 745645
#      sources:
#        - 767
#        - 24789
#      destinations:
#        - 423
#        - 224
#
#
#modules:
## https://wiki.yandex-team.ru/market/development/developer-experience/mj-framework/service.yaml/#modulpostgres
#  postgres:
#    env:
#      local:
#        embedded:
#          enabled: true
#          zonky:
#            port: 5454
#      functionalTestRecipe:
#        embedded:
#          enabled: true
#          type: recipe
## https://wiki.yandex-team.ru/market/development/developer-experience/mj-framework/service.yaml/#modulquartz
#  quartz:
#    env:
#      local:
#        jobStoreType: MEMORY
## https://wiki.yandex-team.ru/market/development/developer-experience/mj-framework/service.yaml/#modulmongo
#  mongo:
#    env:
#      local:
#        embedded:
#          enabled: true
#  bazinga:
#
##arcadia/folder/disable
#generation_settings:
#  ya_make_dependencies: arcadia
#  modules_templates: folder
#  api_service_classes: disable
