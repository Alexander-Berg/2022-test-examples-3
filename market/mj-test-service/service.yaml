clownductor_service_info:
  clownductor_project: market
  preset:
    name: x2nano
  design_review: MARKETDX-168
  description: Тестовый сервис для тестирования Market Java Framework
  duty_group_id: marketdevexp
  has_balancers: true
  release_flow:
    single_approve: true

meta:
  service_name: mj-test-service
  maintainers:
    - Ilya Sapachev <sid-hugo@yandex-team.ru>
  wiki: https://wiki.yandex-team.ru/market/development/developer-experience/mj/

java_service:
  service_name: mj-test-service
  root_package: ru.yandex.market.mj.test.service
  deploy_type: nanny

trace:
  module: FRAMEWORK_TEST_SERVICE

modules:
  postgres:
