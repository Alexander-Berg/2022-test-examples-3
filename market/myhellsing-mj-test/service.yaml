java_service:
  service_name: myhellsing-mj-test
  root_package: ru.yandex.market.marketdevexp
  deploy_type: yandex_deploy

trace:
  module: MARKET_MARKETDEVEXP

#clients:
#  fault_tolerance_presets:
#    circuit_breakers:
#      test_service:
#        failure_rate_threshold: 50
#        permitted_number_of_calls_in_half_open_state: 10
#        sliding_window_size: 100
#        sliding_window_type: COUNT_BASED
#        minimum_number_of_calls: 100
#        writable_stack_trac_enabled: true
#        automatic_transition_from_open_to_half_open_enabled: false
#        wait_interval_function_in_open_state_seconds: 60
#        slow_call_rate_threshold: 100
#        slow_call_duration_threshold_seconds: 60
#        max_wait_duration_in_half_open_state_seconds: 0
#    retries:
#      test_service:
#        max_attempts: 3
#        fail_after_max_attempts: false
#        writable_stack_trace_enabled: true
#        wait_duration_millis: 3000
#
#  list:
#    - openapi_spec_path: market/dev-exp/services/myhellsing-mj-test/src/main/resources/openapi/api/api.yaml
#      circuit_breaker_id: test_service
#      retry_id: test_service

#tvm:
#  sources:
#    - 2029758
#  destinations:
#    - 2029758
#  env:
#    local:
#      serverTvmDisabled: false
#      clientsTvmDisabled: true
#    production:
#      id: 8888
#      sources:
#        - 32343
#        - 645647
#      destinations:
#        - 423
#        - 224
#    testing:
#      id: 7777
#      sources:
#        - 424
#        - 647
#      destinations:
#        - 42213
#        - 2243
#    prestable:
#      id: 745645
#      sources:
#        - 767
#        - 24789
#      destinations:
#        - 423
#        - 224
#
#
#modules:
#  postgres:
#    env:
#      local:
#        embedded:
#          enabled: true
#          zonky:
#            port: 5454
#      functionalTestRecipe:
#        embedded:
#          enabled: true
#          type: recipe
#  quartz:
#    env:
#      local:
#        jobStoreType: MEMORY
#  bazinga:
#  mongo:
#
##arcadia/folder/disable
#generation_settings:
#  ya_make_dependencies: arcadia
#  modules_templates: folder
#  api_service_classes: disable
