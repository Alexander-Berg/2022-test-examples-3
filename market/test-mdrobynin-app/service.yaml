java_service:
  service_name: test-mdrobynin-app
  root_package: ru.yandex.market.testmdrobyninapp
  deploy_type: nanny
  latest_mj_version: true

trace:
  module: MARKET_TEST_MDROBYNIN_APP

#clients:
#  list:
#    self_client:
#      openapi_spec_path: market/dev-exp/test-mdrobynin-app/src/main/resources/openapi/api/api.yaml
#      fault_tolerance:
#        circuit_breaker_id: mj
#        retry_id: mj

#tvm:
#  sources:
#    - 2029758
#  destinations:
#    - 2029758
#  env:
#    local:
#      serverTvmDisabled: false
#      clientsTvmDisabled: true
#    production:
#      id: 8888
#      sources:
#        - 32343
#        - 645647
#      destinations:
#        - 423
#        - 224
#    testing:
#      id: 7777
#      sources:
#        - 424
#        - 647
#      destinations:
#        - 42213
#        - 2243
#    prestable:
#      id: 745645
#      sources:
#        - 767
#        - 24789
#      destinations:
#        - 423
#        - 224
#
#
#modules:
## https://wiki.yandex-team.ru/market/development/developer-experience/mj-framework/service.yaml/#modulpostgres
#  postgres:
#    env:
#      local:
#        embedded:
#          enabled: true
#          zonky:
#            port: 5454
#      functionalTestRecipe:
#        embedded:
#          enabled: true
#          type: recipe
## https://wiki.yandex-team.ru/market/development/developer-experience/mj-framework/service.yaml/#modulquartz
#  quartz:
#    env:
#      local:
#        jobStoreType: MEMORY
## https://wiki.yandex-team.ru/market/development/developer-experience/mj-framework/service.yaml/#modulmongo
#  mongo:
#    env:
#      local:
#        embedded:
#          enabled: true
#  bazinga:
#
##arcadia/folder/disable
#generation_settings:
#  ya_make_dependencies: arcadia
#  modules_templates: folder
#  api_service_classes: disable
