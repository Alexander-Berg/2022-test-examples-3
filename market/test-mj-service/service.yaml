java_service:
  service_name: test-mj-service
  root_package: ru.yandex.market.mini.testmjservice
  deploy_type: nanny
  latest_mj_version: true

trace:
  module: MARKET_TEST_MJ_SERVICE

#clients:
#  list:
#    self_client:
#      openapi_spec_path: market/mini-market/test-mj-service/src/main/resources/openapi/api/api.yaml
#      fault_tolerance:
#        circuit_breaker_id: mj
#        retry_id: mj

#tvm:
#  sources:
#    - 2029758
#  destinations:
#    - 2029758
#  env:
#    local:
#      serverTvmDisabled: false
#      clientsTvmDisabled: true
#    production:
#      id: 8888
#      sources:
#        - 32343
#        - 645647
#      destinations:
#        - 423
#        - 224
#    testing:
#      id: 7777
#      sources:
#        - 424
#        - 647
#      destinations:
#        - 42213
#        - 2243
#    prestable:
#      id: 745645
#      sources:
#        - 767
#        - 24789
#      destinations:
#        - 423
#        - 224
#
#
#modules:
## https://wiki.yandex-team.ru/market/development/developer-experience/mj-framework/service.yaml/#modulpostgres
#  postgres:
#    env:
#      local:
#        embedded:
#          enabled: true
#          zonky:
#            port: 5454
#      functionalTestRecipe:
#        embedded:
#          enabled: true
#          type: recipe
## https://wiki.yandex-team.ru/market/development/developer-experience/mj-framework/service.yaml/#modulquartz
#  quartz:
#    env:
#      local:
#        jobStoreType: MEMORY
## https://wiki.yandex-team.ru/market/development/developer-experience/mj-framework/service.yaml/#modulmongo
#  mongo:
#    env:
#      local:
#        embedded:
#          enabled: true
#  bazinga:
#
##arcadia/folder/disable
#generation_settings:
#  ya_make_dependencies: arcadia
#  modules_templates: folder
#  api_service_classes: disable
