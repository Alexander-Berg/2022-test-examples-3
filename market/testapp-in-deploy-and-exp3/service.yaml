java_service:
  service_name: testapp-in-deploy-and-exp3
  root_package: ru.yandex.market.testappindeployandexp3
  deploy_type: yandex_deploy
  latest_mj_version: true

trace:
  module: MARKET_TESTAPP_IN_DEPLOY_AND_EXP

#clients:
#  list:
#    self_client:
#      openapi_spec_path: market/devexp/testapp-in-deploy-and-exp3/src/main/resources/openapi/api/api.yaml
#      fault_tolerance:
#        circuit_breaker_id: mj
#        retry_id: mj

#tvm:
#  sources:
#    - 2029758
#  destinations:
#    - 2029758
#  env:
#    local:
#      serverTvmDisabled: false
#      clientsTvmDisabled: true
#    production:
#      id: 8888
#      sources:
#        - 32343
#        - 645647
#      destinations:
#        - 423
#        - 224
#    testing:
#      id: 7777
#      sources:
#        - 424
#        - 647
#      destinations:
#        - 42213
#        - 2243
#    prestable:
#      id: 745645
#      sources:
#        - 767
#        - 24789
#      destinations:
#        - 423
#        - 224
#
#
#modules:
## https://wiki.yandex-team.ru/market/development/developer-experience/mj-framework/service.yaml/#modulpostgres
#  postgres:
#    env:
#      local:
#        embedded:
#          enabled: true
#          zonky:
#            port: 5454
#      functionalTestRecipe:
#        embedded:
#          enabled: true
#          type: recipe
## https://wiki.yandex-team.ru/market/development/developer-experience/mj-framework/service.yaml/#modulquartz
#  quartz:
#    env:
#      local:
#        jobStoreType: MEMORY
## https://wiki.yandex-team.ru/market/development/developer-experience/mj-framework/service.yaml/#modulmongo
#  mongo:
#    env:
#      local:
#        embedded:
#          enabled: true
#  bazinga:
#  yav-properties:
#    env:
#      local:
#        enable: true
#
##arcadia/folder/disable
#generation_settings:
#  ya_make_dependencies: arcadia
#  modules_templates: folder
#  api_service_classes: disable
