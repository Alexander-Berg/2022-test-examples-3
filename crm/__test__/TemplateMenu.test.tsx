import React from 'react';
import { render, screen, act, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { TemplatesMenu } from '../TemplatesMenu';
import { TemplateProviderStub } from './TemplateProvider.stub';
import { TemplatesMenuService } from '../TemplatesMenu.service';

const createTestEnv = () => {
  const templateProviderStub = new TemplateProviderStub();
  const templateTypes = templateProviderStub.getTypes();
  const templateStub = templateProviderStub.getTemplate('1');

  const handleTemplateClickMock = jest.fn();
  const getTemplateListByFilterSpy = jest.spyOn(templateProviderStub, 'getTemplateListByFilter');

  const { container } = render(
    <TemplatesMenu
      templatesMenuService={new TemplatesMenuService(new TemplateProviderStub())}
      onTemplateClick={handleTemplateClickMock}
    />,
  );

  getTemplateListByFilterSpy.mockClear();

  return {
    container,
    templateStub,
    templateTypes,
    handleTemplateClickMock,
    getTemplateListByFilterSpy,
  };
};

describe('TemplatesMenu', () => {
  describe('base render with stub', () => {
    it('renders ui', () => {
      const { container } = createTestEnv();

      expect(container).toMatchSnapshot();
    });
  });

  describe('when change template type', () => {
    it('calls getTemplateListByFilter', () => {
      const { templateTypes, getTemplateListByFilterSpy } = createTestEnv();

      act(() => {
        userEvent.click(screen.getByText(templateTypes[1].caption));
      });

      waitFor(() => {
        expect(getTemplateListByFilterSpy).toBeCalledWith({ typeId: templateTypes[1].id });
      });
    });
  });

  describe('when change search text', () => {
    it('calls getTemplateListByFilter', () => {
      const { templateTypes, getTemplateListByFilterSpy } = createTestEnv();

      act(() => {
        userEvent.type(screen.getByPlaceholderText('Поиск'), 'search');
      });

      waitFor(() => {
        expect(getTemplateListByFilterSpy).toBeCalledWith({
          typeId: templateTypes[0].id,
          text: 'search',
        });
      });
    });
  });

  describe('.onTemplateClick', () => {
    it('calls with template', () => {
      const { templateStub, handleTemplateClickMock } = createTestEnv();

      act(() => {
        userEvent.click(screen.getByText(templateStub.name!));
      });

      expect(handleTemplateClickMock).toBeCalledTimes(1);
      expect(handleTemplateClickMock).toBeCalledWith(templateStub);
    });
  });
});
