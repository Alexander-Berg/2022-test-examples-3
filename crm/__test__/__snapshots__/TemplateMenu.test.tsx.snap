// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplatesMenu base render with stub renders ui 1`] = `
<div>
  <div
    class="TemplatesMenu"
  >
    <div
      class="TemplatesMenu__list"
    >
      <span
        class="RadioButton RadioButton_size_s RadioButton_view_default CRMRadioButton_width_max"
        role="radiogroup"
      >
        <div
          class="RadioButton-Plate"
          hidden=""
        />
        <label
          class="RadioButton-Radio RadioButton-Radio_checked"
        >
          <input
            autocomplete="off"
            checked=""
            class="RadioButton-Control"
            name="xuniq-0-1"
            type="radio"
            value="type_1"
          />
          <div
            class="RadioButton-Content"
          >
            Type 1
          </div>
        </label>
        <label
          class="RadioButton-Radio"
        >
          <input
            autocomplete="off"
            class="RadioButton-Control"
            name="xuniq-0-1"
            type="radio"
            value="type_2"
          />
          <div
            class="RadioButton-Content"
          >
            Type 2
          </div>
        </label>
      </span>
      <span
        class="Textinput Textinput_view_default Textinput_size_s"
      >
        <input
          aria-invalid="false"
          autocomplete="off"
          class="Textinput-Control"
          data-testid="TextInput"
          placeholder="Поиск"
          type="text"
          value=""
        />
        <span
          class="Textinput-Box"
        />
      </span>
      <div
        aria-multiselectable="false"
        class="Menu Menu_width_max Menu_size_s Menu_view_default"
        view="default"
        width="max"
      >
        <div
          aria-selected="false"
          class="Menu-Item Menu-Item_type_menuitem"
          data-key="item-0"
          id="item-0-0"
          role="menuitem"
        >
          <span
            class="Menu-Text"
          >
            <span
              class="TemplateMenuItem__overlay"
            />
            template name 1
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
